modules/tinymce/src/core/main/ts/init/InitContentBody.ts

```diff
@@ -453,17 +460,23 @@ const initContentBody = (editor: Editor, skipWrite?: boolean) => {
   Placeholder.setup(editor);
 
   Events.firePreInit(editor);
+  preInit(editor);
 
   Rtc.setup(editor).fold(() => {
-    preInit(editor);
-  }, (loadingRtc) => {
+    Events.firePostRender(editor);
```

Comment:
Changing this event like this is pretty much a breaking change. I don't think we can do this as it's moved a reasonable amount later in the process now and we have no idea how many people were relying on it firing at the old location sorry.

Reply:
For context, a simple example of how this may break an integration is if an integration changed the `directionality` in the `PostRender`. Previously that would have taken effect, but it now no longer will since the directionality is read before `PostRender` fires.

Reply:
Good point let me see if I can move things around so that the rtc instance is setup before preInit and so the we can do all the steps in the exact same order. You can pull the "breaking change card" for any change it's all about how much impact it might have on users. The 5.2, 5.3 change from sync to async init of the content of the editor was far more "breaking" than this re-ordering and it doesn't seem to have broken any integrations. We really need to fix this rendering look at the attached video of the editor blinking it looks really unprofessional. The firePostRender needs to happen early on other wise you get an editor without a toolbar then after a while you get the toolbar and the content so it produces a new type of flash.

Reply:
> You can pull the "breaking change card" for any change it's all about how much impact it might have on users.

Yeah, it has a low risk of breaking someone's integration, however I don't believe we need to do it in this case. We can either change things to make sure the event fires in the same order as you mentioned or look at doing the `hidden` trick I mentioned. If neither of those work though, then yeah we'll have to sort it out and potentially end up changing where things fire.

> The 5.2, 5.3 change from sync to async init of the content of the editor was far more "breaking" than this re-ordering and it doesn't seem to have broken any integrations

I'm going to have to disagree here, as integrations had no hooks into the part of the init sequence that was changed and instead they could only listen for the events we fire. Additionally, the whole init sequence has always been async, since it needs to load stylesheets/javascript files.

Reply:
> I'm going to have to disagree here, as integrations had no hooks into the part of the init sequence that was changed and instead they could only listen for the events we fire. Additionally, the whole init sequence has always been async, since it needs to load stylesheets/javascript files.

Right it was only the RTC part that got converted into an async operation I thought it was both of the branches.

I moved things around now so the postRender event should now fire where it used to fire. Not sure what makes more sense here using a thunk or having a separate function that setups the noop rtcInstance.

Reply:
Thanks Spocke! At a glance that should be fine now, but I'll review it properly in the morning.

Reply:
This would still need the firePostRender before content css loading change since otherwise we still get that half rendered UI while the RTC content is loading I want the full editor UI in readonly mode with a throbber while the content is loading and that is what is happening now. We could change this to use a hidden state on the body but that would still require the other changes to the loading to avoid that for RTC and it would be a bit more involved that for example in inline mode we don't want to toggle the hidden state on/off since that would flash the editor contents while loading the css in and the page css might already have all the styles it need to render the content. So I think we are good with the things as is for now.

Reply:
I think the thunk is fine.

Whether the UI displays with a toolbar or not is less important than the throbber - that must display immediately. If we can order the events (or safely change other parts of the code) such that the toolbar shows, that's cool, if not that's fine too.